### DISCUSSION TITLE
Getting fit stats (e.g. coefficients) when using MLForecast

### DISCUSSION THREAD

**Discussion by FirewingTitan** [Nixtla/mlforecast#181] [Q&A] [answered]
Hi,

I'm trying to figure out how to get the fit statistics following training when using MLForecast. As an example, below I have a simple linear regression model. When I fit it on the data, I can get the coefficients. I'm trying to find out how to get the same when using MLForecast. I checked out fcst.preprocess but looks like that is used before choosing the model.

Reproducible code is below.
--------------------------------

`#### Get the libraries
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
from sklearn.linear_model import LinearRegression

#### Generate the data
rng = np.random.RandomState(90)
serie_length = 7 * 20
dates = pd.date_range('2000-01-01', freq = 'D', periods = serie_length, name='ds')
y = dates.dayofweek + rng.randint(-1, 2, size=dates.size)
data = pd.DataFrame({'y': y.astype(np.float64)}, index=dates)

#### Create the training data
valid_horizon = 14
train = data.head(- valid_horizon).copy()
y_valid = data.tail(valid_horizon)['y']

#### Introduce lag
train['lag-7'] = train['y'].shift(7)
train['lag-14'] = train['y'].shift(14)

#### Drop the null values before referencing.
train_without_nulls = train.dropna()

x_train = train_without_nulls.drop(columns='y')
y_train = train_without_nulls['y']

lr = LinearRegression(fit_intercept = False).fit(x_train, y_train)
lr.coef_
#### ========================
from mlforecast.core import TimeSeries
from mlforecast import MLForecast
from mlforecast import forecast

#### Get the training data
train_mlfcst = train.reset_index()[['ds', 'y']]
train_mlfcst.index = pd.Index(np.repeat(0, train.shape[0]), name = 'unique_id')
train_mlfcst.reset_index(inplace = True)

#### Now we define our simple model.
model = LinearRegression(fit_intercept = False)

#### We create an MLForecast object with the model
fcst = MLForecast(
    models = model,
    lags = [7, 14],
    num_threads = 1,
    freq = 'D'
)

#### Now fit the model.
fcst.fit(
        df = train_mlfcst,
        id_col = 'unique_id',
        time_col = 'ds',
        target_col = 'y'
        )

#### ***************************************************************************

`

**Comment by jmoralez** [comment, marked as answer]
Hey. After you fit the models are available in the models_ attribute, so using `fcst.models_['LinearRegression'].coef_` should work

**Reply by FirewingTitan** [reply]
Hi jmoralex, yes that worked. 

Perhaps an example like this and a bit in the documentation won't be good.

Best wishes.

**Reply by jmoralez** [reply]
Sure, will add a couple more sections soon. Thanks!
